icicle-compiler/src/Icicle/Sea/IO/Psv/Output.hs

```diff
@@ -96,29 +98,29 @@ seaOfWriteFleetOutput config whitelist states = do
     , "}"
     ]
 
-seaOfChordTime :: Mode -> Doc
-seaOfChordTime = \case
+seaOfChordName :: Mode -> Doc
+seaOfChordName = \case
   Snapshot _ -> vsep
-    [ "const char  *chord_time = \"\";"
-    , "const size_t chord_size = 0;"
+    [ "const char  *chord_name = \"\";"
+    , "const size_t chord_name_length = 0;"
     ]
-  Chords     -> vsep
-    [ "iint_t c_year, c_month, c_day, c_hour, c_minute, c_second;"
-    , "itime_to_gregorian (chord_times[chord_ix], &c_year, &c_month, &c_day, &c_hour, &c_minute, &c_second);"
-    , ""
-    , "const size_t chord_size = sizeof (\"|yyyy-mm-ddThh:mm:ssZ\") - 1;"
-    , "char chord_time[chord_size + 1];"
-    , "snprintf (chord_time, chord_size + 1, \"|" <> timeFmt <> "\", "
-             <> "c_year, c_month, c_day, c_hour, c_minute, c_second);"
+  Chords -> vsep
+    [ "const int64_t  chord_name_offset = chord_name_offsets[chord_ix];"
+    , "const int64_t  chord_name_length = chord_name_lengths[chord_ix];"
+    , "const char    *chord_name = (char*) (chord_name_data + chord_name_offset);"
     ]
 
 timeFmt :: Doc
 timeFmt = "%04\" PRId64 \"-%02\" PRId64 \"-%02\" PRId64 \"T%02\" PRId64 \":%02\" PRId64 \":%02\" PRId64 \"Z"
 
 outputChord :: Doc
 outputChord
-  = outputValue "string" ["chord_time", "chord_size"]
-
+  = vsep
+  [ "if (chord_name_length != 0) {"
```

Comment:
It might be better to have this as `chord_name != NULL` and change the code above to set `chord_name = NULL`  in the snapshot case.

It probably won't happen, but the empty string seems like a perfectly valid chord name, so it would be odd if that caused the column to be dropped off.